spock-specs/src/test/groovy/org/spockframework/smoke/ast/condition/BaseVerifyMethodsAstSpec.groovy

```diff
@@ -0,0 +1,215 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.spockframework.smoke.ast.condition
+
+import org.spockframework.EmbeddedSpecification
+import org.spockframework.specs.extension.SpockSnapshotter
+import spock.lang.Snapshot
+
+import java.lang.annotation.Annotation
+
+abstract class BaseVerifyMethodsAstSpec extends EmbeddedSpecification {
+
+  @Snapshot(extension = 'groovy')
+  SpockSnapshotter snapshotter
+
+  abstract Class<? extends Annotation> getAnnotation()
+
+  def "transforms conditions in methods inside spec classes"() {
+    given:
+    snapshotter.specBody()
+
+    when:
+    def result = compiler.transpileSpecBody("""
+@${annotation.name}
+void isPositiveAndEven(int a) {
+    a > 0
+    a % 2 == 0
+}
+""")
+
+    then:
+    snapshotter.assertThat(result.source).matchesSnapshot()
+  }
+
+  def "transforms conditions in methods outside of spec classes"() {
+    when:
+    def result = compiler.transpile("""
+class Assertions {
+    @${annotation.name}
+    void isPositiveAndEven(int a) {
+        a > 0
+        a % 2 == 0
+    }
+}
+""")
+
+    then:
+    snapshotter.assertThat(result.source).matchesSnapshot()
+  }
+
+  def "transforms conditions in private methods"() {
+    given:
+    snapshotter.specBody()
+
+    when:
+    def result = compiler.transpileSpecBody("""
+@${annotation.name}
+private void isPositive(int a) {
+    a > 0
+}
+""")
+
+    then:
+    snapshotter.assertThat(result.source).matchesSnapshot()
+  }
+
+  def "transforms conditions in static methods"() {
+    when:
+    def result = compiler.transpile("""
+class Assertions {
+    @${annotation.name}
+    static void isPositive(int a) {
+        a > 0
+    }
+}
+""")
+
+    then:
+    snapshotter.assertThat(result.source).matchesSnapshot()
+  }
+
+  def "transforms explicit conditions"() {
+    when:
+    def result = compiler.transpile("""
+class Assertions {
+    @${annotation.name}
+    static void isPositiveAndEven(int a) {
+        assert a > 0
+        assert a % 2 == 0
+    }
+}
+""")
+
+    then:
+    snapshotter.assertThat(result.source).matchesSnapshot()
+  }
+
+  def "methods without condition declarations stay unchanged"() {
+    when:
+    def result = compiler.transpile("""
+class NonAssertions {
+    @${annotation.name}
+    static void copy(Integer a) {
+        def uselessCopy = a
+    }
+}
+""")
+
+    then:
+    snapshotter.assertThat(result.source).matchesSnapshot()
+  }
+
+  def "ignores methods without annotation"() {
```

Comment:
Why did you choose to ignore it instead of throwing a compilation error?

Reply:
Why would we throw an exception here? It would mean that a class can only have methods annotated with `@Verify`? Are you asking about a different scenario?

Reply:
But you are right, it probably makes sense to fail if the method has a non-void return type. Changed it in [`6875970` (#2112)](https://github.com/spockframework/spock/pull/2112/commits/68759709b199b91bc7a5fe462918e9bfaf959483)